src/java/org/apache/cassandra/index/sai/disk/vector/OnDiskOrdinalsMap.java

```diff
@@ -259,6 +270,26 @@ public int getOrdinalForRowId(int rowId) throws IOException
             return reader.readInt();
         }
 
+        @Override
+        public void forEachOrdinalInRange(int startRowId, int endRowId, OrdinalConsumer consumer) throws IOException
+        {
+            reader.seek(rowOrdinalOffset);
+            // sequential read without seeks should be fast, we expect OS to prefetch data from the disk
+            // binary search for starting offset of min rowid >= startRowId unlikely to be faster
+            for (long idx = 0; idx < high; idx ++)
+            {
+                int rowId = reader.readInt();
```

Comment:
I think we need to verify that we can read the next int.

Reply:
"high" is calculated on base of segmentEnd, not to mention that "getOrdinalForRowId()" doesn't do such checks

Reply:
You're right, I missed that detail.